File: Golang/53.最大子数组和.go
package leetcode

func maxSubArray(nums []int) int {
	last, ans := 0, nums[0]
	for _, v := range nums {
		last = maxInt(last+v, v)
		ans = maxInt(ans, last)
	}
	return ans
}

func maxInt(x, y int) int {
	if x > y {
		return x
	}
	return y
}

File: 53.最大子数组和.cc
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int last = 0, ans = nums[0];
        for (const int& x : nums) {
            last = max(last + x, x);
            ans = max(ans, last);
        }
        return ans;
    }
};